### PR TITLE
Try to recover when encountering JPEG markers with too short marker lengths (issue 8169)

### LINKED ISSUE #8169: Error: JPEG error: unknown marker 2e00
Link to problematic PDF: [prob1.pdf](http://210.243.166.143/prob1.pdf)
Link to pdf.js we use (I zip mozillaPDF folder into a zip): [mpdf.zip](http://210.243.166.143/mpdf.zip)

Configuration:
- Web browser and its version: _Chrome v44_
- Operating system and its version: _Win7 Pro 64bit_
- PDF.js version: _1.6.230_
- Is an extension: _not sure_

Steps to reproduce the problem:
1. Just open it using pdf.js and see error messsage in console.
2.All other pdf reader s/w or browser plugin CAN sohw image well.

What is the expected behavior? (add screenshot) Should show a single image in pdf.js

What went wrong? (add screenshot) show blank in pdf.js aftrer pdf downloaded. browser console shows:"
PDF e125b57ca39f9091e0558b41122f6a89 [1.3 Image2PDF (http://www.verypdf.com) / Image2PDF Command Line Software] (PDF.js: 1.6.230) viewer.js:8269
Error: JPEG error: unknown marker 2e00 pdf.js:10392
at error (http://un.udndata.com/mozillaPDF/build/pdf.worker.js:1299:17)
at Object.parse (http://un.udndata.com/mozillaPDF/build/pdf.worker.js:11971:13)
at Object.JpegStream_ensureBuffer [as ensureBuffer] (http://un.udndata.com/mozillaPDF/build/pdf.worker.js:20747:15)
at Object.JpegStream_getBytes [as getBytes] (http://un.udndata.com/mozillaPDF/build/pdf.worker.js:20756:10)
at Object.PDFImage_getImageBytes [as getImageBytes] (http://un.udndata.com/mozillaPDF/build/pdf.worker.js:34560:25)
at Object.PDFImage_createImageData [as createImageData] (http://un.udndata.com/mozillaPDF/build/pdf.worker.js:34435:27)
at Object.onResolve (http://un.udndata.com/mozillaPDF/build/pdf.worker.js:37363:34)
at Object.runHandlers (http://un.udndata.com/mozillaPDF/build/pdf.worker.js:2250:35) pdf.js:10392
Warning: Unable to decode image: Error: JPEG error: unknown marker 2e00 pdf.js:10392
Warning: Dependent image isn't ready yet
Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
We embed pdf.js in our web page but that's paid-service pages, so I put pdf.js and the pdf.js (zip mozillaPDF folder as a zip file) on web server as public links for you tech guys to analazye.

### PR DESCRIPTION
**Pull request by Snuffleupagus**
The issue with the JPEG image in question, is that the COM (Comment) marker has an incorrect length entry.

Fixes #8169.

### CODE REVIEW

File: src/core/jpg.js [hunk 1]
```diff
@@ -824,6 +824,12 @@ var JpegImage = (function JpegImageClosure() {
               // block was eaten by the encoder
               offset -= 3;
               break;
+            } else if (data[offset] === 0xFF &&
+                       data[offset + 1] >= 0xC0 && data[offset + 1] <= 0xFE) {
+              // If the length of the previous block is incorrect (too short),
+              // treat the current "marker" as padding and skip past it when
+              // the next marker is a valid one (fixes issue8169.pdf).
+              break;
```
Comment (yurydelendik) [inline src/core/jpg.js]: I think it was a "luck" that there is no other data in-between. We need to add a check to the readDataBlock and if there is no marker after the read block, throw a warning and seek a next marker starting from start of the read block.

Comment (Snuffleupagus) [inline src/core/jpg.js]: Good point; is the updated patch better?
*Note:* I've not had time to run all tests locally yet, so if the approach is OK I'll trigger testing later.

**Review by yurydelendik — APPROVED** [review]
Looks good

**Pull request merged**